### PR TITLE
Mesos HTTP Adaptor

### PR DESCRIPTION
**Pull request by mohitsoni**
This PR migrates SDK to use Mesos HTTP Adaptor.

By default, with Mesos HTTP Adaptor frameworks will continue to use V0 API of Mesos. Optionally, an operator can set `MESOS_API_VERSION` environment to `V1` to make framework use Mesos HTTP API.

For more details refer: https://github.com/mesosphere/mesos-http-adapter

### CODE REVIEW

**Comment by mohitsoni** [comment]
@mesosphere-ci retest this please.

**Comment by triclambert** [comment]
What about universe plumbing for the env var?

**Comment by mohitsoni** [comment]
@triclambert We need to plumb that on a per-framework basis inside framework's own universe package. 

And, because we are using a monorepo, let me wire this feature inside the `helloworld` and `kafka` frameworks.

**Comment by mohitsoni** [comment]
The PR fails to deploy currently due to:

```
Exception in thread "main" java.lang.UnsatisfiedLinkError: org.apache.mesos.v1.scheduler.V0Mesos.initialize()V
```

We need to get libmesos for `1.1.0` uploaded to downloads, and configure the frameworks to use that.

**Comment by mohitsoni** [comment]
@mesosphere-ci retest this please

**Comment by gabrielhartmann** [comment]
@mohitsoni:  Now that we have the libmesos-bundle, this shim can be plausibly merged correct?

**Comment by mohitsoni** [comment]
@mesosphere-ci retest this please

**Comment by mohitsoni** [comment]
@mesosphere-ci retest this please

File: frameworks/hdfs/universe/resource.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "assets": {
     "uris": {
       "jre-tar-gz": "https://downloads.mesosphere.com/java/jre-8u112-linux-x64.tar.gz",
-      "libmesos-bundle-tar-gz": "http://downloads.mesosphere.com/libmesos-bundle/libmesos-bundle-1.8.7-1.0.2-2.tar.gz",
+      "libmesos-bundle-tar-gz": "https://s3.amazonaws.com/downloads.mesosphere.io/libmesos-bundle/libmesos-bundle-1.9-argus-1.1.x-2.tar.gz",
```
Comment (triclambert) [inline frameworks/hdfs/universe/resource.json]: Why not downloads.mesosphere.com?

Comment (mohitsoni) [inline frameworks/hdfs/universe/resource.json]: 👍

File: frameworks/hdfs/universe/resource.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "assets": {
     "uris": {
       "jre-tar-gz": "https://downloads.mesosphere.com/java/jre-8u112-linux-x64.tar.gz",
-      "libmesos-bundle-tar-gz": "http://downloads.mesosphere.com/libmesos-bundle/libmesos-bundle-1.8.7-1.0.2-2.tar.gz",
+      "libmesos-bundle-tar-gz": "https://downloads.mesosphere.io/libmesos-bundle/libmesos-bundle-1.9-argus-1.1.x-2.tar.gz",
```
Comment (triclambert) [inline frameworks/hdfs/universe/resource.json]: @lingmann is there any reason to prefer .com or .io?

**Comment by mohitsoni** [comment]
@mesosphere-ci retest this please

**Pull request merged**